### DISCUSSION TITLE
Difference between SPEL and Evrete

### DISCUSSION THREAD

**Discussion by kaggrwal** [evrete/evrete#56] [Q&A]
Hi,

I'm really impressed by evrete library, infact I'm using it in my own project. When i was exploring the solutions i came across [SPEL](https://docs.spring.io/spring-framework/docs/3.0.x/reference/expressions.html).

This was earlier used in one of our projects, a rule engine was created with this. So, i wonder what are major differences between both. One difference i can clearly see is that we can use stateful sessions with evrete. 
I know this can be a beginner question but i wanted to know the views from the community 🙏

**Comment by andbi** [comment]
Hi! 

In fact, SPEL is just a scripting language having nothing to do with rule engines. Evrete has very basic support for referencing object's properties inside literal conditions, but all such expressions are immediately resolved and compiled into Java bytecode.

**Comment by kaggrwal** [comment]
Hi @andbi,

Thanks for answering this, I understand that SPEL is just an expression language. 
Our old rule engine is mostly developed on this [concept](https://medium.com/@er.rameshkatiyar/implement-your-own-rule-engine-java8-springboot-mvel-5928474e1ba5).

So, I was looking for all the flexibility and advantages that Evrete is providing over this so that it would be easier for me to prepare some kind of concise documentation/blog.
